homeassistant/components/stream/worker.py

```diff
@@ -17,32 +20,41 @@
     PACKETS_TO_WAIT_FOR_AUDIO,
     SEGMENT_CONTAINER_FORMAT,
     SOURCE_TIMEOUT,
+    TARGET_PART_DURATION,
 )
-from .core import Segment, StreamOutput
-from .fmp4utils import get_init_and_moof_data
+from .core import Part, Segment, StreamOutput
 
 _LOGGER = logging.getLogger(__name__)
 
 
 class SegmentBuffer:
     """Buffer for writing a sequence of packets to the output as a segment."""
 
-    def __init__(self, outputs_callback) -> None:
+    def __init__(
+        self, outputs_callback: Callable[[], Mapping[str, StreamOutput]]
+    ) -> None:
         """Initialize SegmentBuffer."""
-        self._stream_id = 0
-        self._outputs_callback = outputs_callback
-        self._outputs: list[StreamOutput] = []
+        self._stream_id: int = 0
+        self._outputs_callback: Callable[
+            [], Mapping[str, StreamOutput]
+        ] = outputs_callback
+        # Each element is a StreamOutput
+        self._outputs: ValuesView[StreamOutput] | list[StreamOutput] = []
         # sequence gets incremented before the first segment so the first segment
         # has a sequence number of 0.
         self._sequence = -1
-        self._segment_start_pts = None
+        self._segment_start_pts: int = cast(int, None)
         self._memory_file: BytesIO = cast(BytesIO, None)
         self._av_output: av.container.OutputContainer = None
         self._input_video_stream: av.video.VideoStream = None
         self._input_audio_stream = None  # av.audio.AudioStream | None
         self._output_video_stream: av.video.VideoStream = None
         self._output_audio_stream = None  # av.audio.AudioStream | None
         self._segment: Segment = cast(Segment, None)
+        self._part_start_dts: int = cast(int, None)
+        self._last_packet_dts: int = cast(int, None)
+        self._part_is_independent = False
```

Comment:
Independent just means that the fragment has a keyframe somewhere in it. Most frames are not keyframes which means they are essentially deltas. Most fragments will not have a keyframe anywhere in them. Finding a fragment with a keyframe is useful because that means a player can start playing from that fragment (without a keyframe it only has deltas with nothing to reference them against).
We don't end up using the independent tag for anything internally, but we will end up passing the information along through the HLS playlist.

Reply:
Sure, let's change it to self._part_has_keyframe